dockerfiles/learn2code_base/Dockerfile

```diff
@@ -11,31 +12,33 @@ RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 # Create the user
 RUN groupadd --gid $USER_GID $USERNAME \
     && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME \
-    #
     # [Optional] Add sudo support. Omit if you don't need to install software after connecting.
     && apt-get update \
     && apt-get install -y sudo \
     && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
     && chmod 0440 /etc/sudoers.d/$USERNAME
 
-# install everythin we need
+# install everything we need
 RUN \
     apt-get -y update \
-    && apt-get install -y git build-essential lldb gdb ninja-build libgtest-dev wget curl software-properties-common \
-    # bring in latest cmake
-    && wget -qO /etc/apt/trusted.gpg.d/kitware-key.asc https://apt.kitware.com/keys/kitware-archive-latest.asc \ 
-    && echo "deb https://apt.kitware.com/ubuntu/ focal main" | tee /etc/apt/sources.list.d/kitware.list \
-    && add-apt-repository -y ppa:system76/pop \
-    && apt-get update \
-    && apt-get install -y cmake \
-    # bring in latest clang toolchain
-    && wget https://apt.llvm.org/llvm.sh \
+    && apt-get install -y git build-essential gdb ninja-build libgtest-dev wget curl software-properties-common libssl-dev\
+    && rm -rf /var/lib/apt/lists/* 
+
+# bring in latest clang toolchain
+RUN \
+    wget https://apt.llvm.org/llvm.sh \
     && chmod +x llvm.sh \
     && ./llvm.sh 16 all \
-    && rm llvm.sh \
-    # clean up
-    && rm -rf /var/lib/apt/lists/*
+    && rm llvm.sh
 
+# bring in latest cmake - download it as binary
+RUN \
+    wget https://github.com/Kitware/CMake/releases/download/v$CMAKE_VERSION/cmake-$CMAKE_VERSION-linux-x86_64.sh \
+    && chmod +x cmake-$CMAKE_VERSION-linux-x86_64.sh \
+    && ./cmake-$CMAKE_VERSION-linux-x86_64.sh --skip-license --include-subdir --prefix=/opt \
```

Comment:
I ran into some issues with having cmake not being installed in `/usr` when using a `.devcontainer` how did you solve this?

Reply:
In the next line a symbolic link is created in */bin*

Since */bin* is also a sym link to */usr/bin* it should work out of the box. 
In case you are using a non ubuntu os, this might be different. 
Just checked it with a .devcontainer and it worked out of the box